htmresearch/algorithms/faulty_temporal_memory.py

```diff
@@ -24,7 +24,7 @@
 """
 import numpy
 from collections import defaultdict
-from nupic.research.temporal_memory import TemporalMemory
+from temporal_memory_phases import TemporalMemory
```

Comment:
Can you do a fully qualified import here? Like `from htmresearch.algorithms.temporal_memory_phases import TemporalMemory`?

The implicit relative import style that you use here is problematic. There is an explicit relative import using a `.` prefix but absolute imports are the best option in most cases.